sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/storage/GcsFileSystem.java

```diff
@@ -93,6 +128,20 @@ protected void copy(List<GcsResourceId> srcResourceIds, List<GcsResourceId> dest
     options.getGcsUtil().copy(toFilenames(srcResourceIds), toFilenames(destResourceIds));
   }
 
+  private List<MatchResult> matchGlobs(List<GcsPath> globs) {
+    return FluentIterable.from(globs)
+        .transform(new Function<GcsPath, MatchResult>() {
+          @Override
+          public MatchResult apply(GcsPath gcsPath) {
+            try {
+              return expand(gcsPath);
+            } catch (IOException e) {
+              return MatchResult.create(Status.ERROR, e);
+            }
+          }})
+        .toList();
```

Comment:
Assuming I'm right, can you please file a JIRA and put a TODO here (and on MatchNonGlobs?) ?